app/src/hid_listener.c

```diff
@@ -78,9 +132,19 @@ int hid_listener(const zmk_event_t *eh) {
         } else {
             hid_listener_keycode_released(ev);
         }
+    } else {
+        const struct zmk_mouse_state_changed *ev = as_zmk_mouse_state_changed(eh);
```

Comment:
not  big fan of this shadowing of `ev`.  Consider
```
    struct zmk_keycode_state_changed *ev = NULL;
    ev = as_zmk_keycode_state_changed(eh);
    if (ev) {
        ...
         return 0;
    }
    ev = as_zmk_mouse_state_changed(eh);
    if (ev) {
        ...
        return 0;
    }
    return 0;
```

Reply:
This doesn't take the fact the events have different types into account. I've implemented the refactor a bit differently, is it better now?